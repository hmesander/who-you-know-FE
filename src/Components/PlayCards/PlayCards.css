.difficulty-button-container {
  width: 80%;
  margin: 3em auto;
  display: flex;
  justify-content: space-evenly;
}

.difficulty-buttons {
  width: 20%;
}