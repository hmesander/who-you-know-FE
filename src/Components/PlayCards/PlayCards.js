import React, { Component } from 'react';
import mockData from '../../helpers/mockData.js';
import { retrieveCardsToPlay } from '../../helpers/apiCalls.js';
import GameBoard from '../GameBoard/GameBoard.js';

export default class PlayCards extends Component {
  constructor() {
    super();

    this.state = {
      difficulty: '',
      cards: [],
      sorted: [],
      score: 0
    };
  }

  setDifficulty = (event) => {
    this.setState({
      difficulty: event.target.value
    });
    this.fetchCardsToPlay(event.target.value);
  }

  fetchCardsToPlay = async (difficulty) => {
    let results = await retrieveCardsToPlay(1, difficulty);
    this.setState({
      cards: mockData
    });
    return results;
  }

  render() {
    if (!this.state.difficulty) {
      return (
        <div>
          <button className='easy-button'onClick={(event) => this.setDifficulty(event)} value='easy'>Easy</button>
          <button className='medium-button' onClick={(event) => this.setDifficulty(event)} value='medium'>Medium</button>
          <button className='hard-button' onClick={(event) => this.setDifficulty(event)} value='hard'>Hard</button>
        </div>
      );
    } else {
      return (
        <div>
          <GameBoard cardsToPlay={this.state.cards} />
        </div>
      );
    } 
  }
}