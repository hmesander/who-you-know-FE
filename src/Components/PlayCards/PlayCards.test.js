/*eslint-disable no-undef camelcase*/

import React from 'react';
import PlayCards from './PlayCards';
import { shallow } from 'enzyme';

describe('PLAYCARDS TESTS', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<PlayCards />);
  });

  it('should match the snapshot when there is a gamedeck to play', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('set the difficulty level in state when set difficulty is called', () => {
    const mockEvent = {target: {value: 'easy'}};
    window.fetch = jest.fn().mockImplementation(() => Promise.resolve({
      json: () => Promise.resolve({
        cards: [{first_last_name: 'Sterling Archer'}]
      })
    }));
    wrapper.instance().setDifficulty(mockEvent);
    const results = wrapper.instance().state.difficulty;

    expect(results).toEqual('easy');
  });

  it('should call set difficulty when the easy button is clicked', () => {
    const mockEvent = { target: { value: 'easy' } };
    const spy = spyOn(wrapper.instance(), 'setDifficulty');

    const button = wrapper.find('.easy-button');
    button.simulate('click', mockEvent);
    expect(spy).toHaveBeenCalled();
  });

  it('should call set difficulty when the medium button is clicked', () => {
    const mockEvent = { target: { value: 'medium' } };
    const spy = spyOn(wrapper.instance(), 'setDifficulty');

    const button = wrapper.find('.medium-button');
    button.simulate('click', mockEvent);
    expect(spy).toHaveBeenCalled();
  });

  it('should call set difficulty when the hard button is clicked', () => {
    const mockEvent = { target: { value: 'hard' } };
    const spy = spyOn(wrapper.instance(), 'setDifficulty');

    const button = wrapper.find('.hard-button');
    button.simulate('click', mockEvent);
    expect(spy).toHaveBeenCalled();
  });
});