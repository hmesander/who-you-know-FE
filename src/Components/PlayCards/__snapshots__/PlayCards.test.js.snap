// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PLAYCARDS TESTS should match the snapshot when there is no username 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <PlayCards />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <h6
        className="login-error-text"
>
        You must be logged in to Play!
</h6>,
      "className": "login-error-container",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "You must be logged in to Play!",
        "className": "login-error-text",
      },
      "ref": null,
      "rendered": "You must be logged in to Play!",
      "type": "h6",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <h6
          className="login-error-text"
>
          You must be logged in to Play!
</h6>,
        "className": "login-error-container",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "You must be logged in to Play!",
          "className": "login-error-text",
        },
        "ref": null,
        "rendered": "You must be logged in to Play!",
        "type": "h6",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
